job-service-admin/src/main/java/com/hpe/caf/services/admin/HibernateUtil.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2015-2017 EntIT Software LLC, a Micro Focus company.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hpe.caf.services.admin;
+
+import org.hibernate.SessionFactory;
+import org.hibernate.boot.registry.StandardServiceRegistryBuilder;
+import org.hibernate.cfg.Configuration;
+import org.hibernate.service.ServiceRegistry;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Objects;
+import java.util.concurrent.locks.Lock;
+import java.util.concurrent.locks.ReentrantLock;
+
+public class HibernateUtil
+{
+
+    private static Logger logger = LoggerFactory.getLogger(HibernateUtil.class);
+
+    private static final String JDBC_PREFIX = "jdbc:";
+
+    private static SessionFactory sessionFactory;
+
+    private static Lock lock = new ReentrantLock();
+
+    /**
+     * Get the single instance of the SessionFactory object for the application's properties.
+     *
+     * Usually an application has a single SessionFactory instance and threads
+     * servicing client requests obtain Session instances from this factory.
+     *
+     * There will be one session factory per application.
+     *
+     * @return SessionFactory
+     */
+    public static SessionFactory getSessionFactory()
```

Comment:
Why are we bring Hibernate into this?  Don't we just use JDBC directly in the web service itself?  What advantage are we getting by bringing Hibernate in here?  Is it to do with connection pooling?